### ISSUE TITLE
Cursor blinks over placeholder text in files when pressing Ctrl+l to search (EOS 6 Beta)

### ISSUE THREAD

**Issue by KarkanAlzwayed** [elementary/files#1711]
This is more of a styling issue. I have noticed that when I hit Ctrl+l to perform a search in files that the cursor starts blinking right in the middle of the letter S of the placeholder text of the search box where it says "Search or Type Path". 

## Expected Behavior
Placeholder text should be visible and the blinking cursor should be blinking before the S, not on it.

## Steps to Reproduce
Open files, hit Ctrl+l, delete the text there to show the place holder text, then you'll see the cursor blinking in the middle of the letter S.

**Comment by jeremypw** [comment]
Yes, I doubt this is anything to do with Files directly.  I use UK locale and keyboard and `<Ctrl>F` to search (on elementary 6).  I find that the cursor appears in the expected position.
![SearchCursor (1)](https://user-images.githubusercontent.com/10513844/118355221-cfa63500-b566-11eb-99b3-dbce47765597.png)

What locale/keyboard do you use?

**Comment by KarkanAlzwayed** [comment]
> Yes, I doubt this is anything to do with Files directly. I use UK locale and keyboard and `<Ctrl>F` to search (on elementary 6). I find that the cursor appears in the expected position.
> ![SearchCursor (1)](https://user-images.githubusercontent.com/10513844/118355221-cfa63500-b566-11eb-99b3-dbce47765597.png)
> 
> What locale/keyboard do you use?

I am using the US keyboard. The cursor on my end is right in the middle of the letter S.

**Comment by jeremypw** [comment]
Do you see this for any other kind of entry with a placeholder? Interestingly, I see in e.g. System Settings that the placeholder only appears when the entry does not have focus (and therefore no cursor).  That would be an easy fix if that is the correct behaviour but it seems a bit odd to me - the user does not so easily see the placeholder if the entry is the default focus.

**Comment by jeremypw** [comment]
@danrabbit What is the correct behaviour for entries? Should the placeholder appear when they are empty or only when they are empty and do not have focus?

**Comment by jeremypw** [comment]
I think the Files Search entry is a bit unusual as it is always the focus when it is visible.  Other entries in elementary apps tend not to be focussed unless they are clicked on.

**Comment by KarkanAlzwayed** [comment]
> @danrabbit What is the correct behaviour for entries? Should the placeholder appear when they are empty or only when they are empty and do not have focus?

I've done some html and CSS before, and I think the placeholder should disappear when the searchbox has been clicked.

**Comment by KarkanAlzwayed** [comment]
> Do you see this for any other kind of entry with a placeholder? Interestingly, I see in e.g. System Settings that the placeholder only appears when the entry does not have focus (and therefore no cursor). That would be an easy fix if that is the correct behaviour but it seems a bit odd to me - the user does not so easily see the placeholder if the entry is the default focus.

I've actually searched throughout the system to find another place that shows this beside files, but it appears to only files (so far) is doing this. Other parts show the placeholder text, but the text disappears when the box is clicked.

**Comment by jeremypw** [comment]
> I've done some html and CSS before, and I think the placeholder text should disappear when the searchbox has been clicked.

The problem with that is it would require an extra click from the user. At the moment you can just start typing after pressing `<Ctrl>F`. In fact you can just start typing at the view and it will switch to search mode automatically.

A placeholder is necessary as the entry is dual purpose and it is not obvious that you can either type a search term or a url.

**Comment by jeremypw** [comment]
A work around would be to add an extra space at the start of the (translated) placeholder in the code.

**Comment by KarkanAlzwayed** [comment]
> A work around would be to add an extra space at the start of the (translated) placeholder in the code.

Thank you. This isn't bothering me at all. Just reporting a small style "issue" since I've volunteered to test the beta on my main machine.

**Comment by jeremypw** [comment]
Normally in Gtk.Entry, placeholder text appears only when not focused - In Files it is drawn especially so maybe it should be offset slightly so as not to interfere with the cursor.  However, best left to the new Gtk4 pathbar to deal with this.
